### PR TITLE
[Term Entry] Dart list method: .indexWhere()

### LINKED ISSUE #4308: [Term Entry] Dart List .indexWhere()
### Reason/inspiration (optional)

We would like a new term entry under Dart [List](https://www.codecademy.com/resources/docs/dart/concept/list) for the Dart List `.indexWhere()` method. 
Provide the entry in a markdown file under `docs/content/dart/concept/list/terms/indexWhere/indexWhere.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by cysiupe**
### Description

- Added the List.indexWhere() term for Dart
- Closes #4308

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by Sriparno08** [comment]
Hey @cysiupe, please sign the CLA by clicking on the 'Contributor License Agreement' link provided in the message from @CLAassistant above as it is a must for first-time contributors.

File: content/dart/concepts/list/terms/indexWhere/indexWhere.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+In Dart, the **`.indexWhere()`** method is for find the index of the first element in a list that fulfill a given condition.
+
+## Syntax
+
+```pseudo
+int indexWhere(bool Function(S) demo)
```
Comment (Sriparno08) [inline content/dart/concepts/list/terms/indexWhere/indexWhere.md:21]:
```suggestion
int indexWhere(bool func(S demo))
```

File: content/dart/concepts/list/terms/indexWhere/indexWhere.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+```
+
+- `int` is the return type of the method.
+- `indexWhere` name of the method.
+- `bool Function(S) demo` it's a function that takes an element of type S as input and returns a boolean value.
+- `S` is a generic type parameter and it represents the type of elements contained in the list. For example if you have a `List<String>`, `S` would be replaced with String.
+- `bool` specifies the return type. In this example it indicates that the function should return a boolean value. Function will return true if the condition is satisfied, false if the function is not fulfilled.
```
Comment (Sriparno08) [inline content/dart/concepts/list/terms/indexWhere/indexWhere.md:28]:
```suggestion
- `func`: A function that provides a condition for the `.indexWhere()` method to check.
- `S`: The type of element that is passed to the function.
- `demo`: The element that is passed to the function.
```

File: content/dart/concepts/list/terms/indexWhere/indexWhere.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Example
+
+In this example we will find index of the first negative number in a list of integers:
```
Comment (Sriparno08) [inline content/dart/concepts/list/terms/indexWhere/indexWhere.md:32]:
```suggestion
In this example, the `.indexWhere()` method is used to find the index of the first negative integer in a list:
```

File: content/dart/concepts/list/terms/indexWhere/indexWhere.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  List<int> numbers = [8, 5, 2, -3, 6];
+
+  int index = numbers.indexWhere((number) => number < 0);
+  if (index != -1) {
+    print('Index of the first negative number: $index');
+  } else {
+    print('No negative numbers found.');
+  }
+}
+```
```
Comment (Sriparno08) [inline content/dart/concepts/list/terms/indexWhere/indexWhere.md:45]:
````suggestion
```dart
void main() {
  List<int> numbers = [8, 5, 2, -3, 6];

  int index = numbers.indexWhere((number) => number < 0);

  if (index != -1) {
    print('Index of the first negative integer: $index');
  } else {
    print('No negative integers found.');
  }
}
```
````

File: content/dart/concepts/list/terms/indexWhere/indexWhere.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+The code returns the fallowing output:
+
+```shell
+"Index of the first negative number: 3"
+```
```
Comment (Sriparno08) [inline content/dart/concepts/list/terms/indexWhere/indexWhere.md:51]:
````suggestion
The above example returns the following output:

```shell
Index of the first negative integer: 3
```
````

**Review by Sriparno08 — COMMENTED** [review]
Hey @cysiupe, thanks for contributing to Codecademy!

I've listed all the modifications that you need to make in your entry below:

**Comment by cysiupe** [comment]
> Hey @cysiupe, thanks for contributing to Codecademy!
> 
> I've listed all the modifications that you need to make in your entry below:

Hi @Sriparno08 . Thank you so much for your help. I applied all requested modifications and I hope this time everything will be fine.

**Review by Sriparno08 — APPROVED** [review]
Looks good for a second review 👍

**Comment by cysiupe** [comment]
> Looks good for a second review 👍

Thank you so much for your help and review @Sriparno08 .

**Review by avdhoottt — APPROVED** [review]
Looks good to get merged!!

**Pull request merged**